### ISSUE TITLE
otel: make sure everything existing actually WAI

### ISSUE THREAD

**Issue by deniseli** [TBD54566975/ftl#2021]
Code Links:
* otelInterceptor: [internal/rpc/context.go#L83](https://github.com/TBD54566975/ftl/blob/main/internal/rpc/context.go#L83)
* otel client: [internal/observability/client.go#L36-L75](https://github.com/TBD54566975/ftl/blob/main/internal/observability/client.go#L36-L75)
* [templates/ftl-controller.yml#L50](https://github.com/tbdeng/ftl-infra/blob/main/templates/ftl-controller.yml#L50) - Exporter is how we send data to otel ([docs](https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/))

Out of scope: this logging interface we provide through the Go runtime SDK that no one currently uses: [go-runtime/ftl/observability/metrics.go](https://github.com/TBD54566975/ftl/blob/main/go-runtime/ftl/observability/metrics.go)

Datadog:
* [go/tbddd](http://go/tbddd) - all TBD services
* [FTL Dashboard](https://app.datadoghq.com/dashboard/39r-qfe-r4w/ftl?fromUser=false&refresh_mode=sliding&view=spans&from_ts=1720457691950&to_ts=1720544091950&live=true)  
* [Controller logs](https://app.datadoghq.com/logs?query=env%3A%22development%22%20service%3A%22ftl-controller%22%20source%3A%22ftl-controller%22%20status%3A%22error%22%20%22failed%20to%20receive%20notification%20conn%20closed%22%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1720543339958&to_ts=1720544239958&live=true)
* Example of a [call](https://app.datadoghq.com/apm/traces?query=env%3Adevelopment%20AND%20service%3A%28ftl-runner%20OR%20ftl-controller%29%20AND%20operation_name%3Aconnectrpc.com_otelconnect.client&cols=core_service%2Ccore_resource_name%2Clog_duration%2Clog_http.method%2Clog_http.status_code&fromUser=false&graphType=flamegraph&historicalData=true&messageDisplay=inline&query_translation_version=v0&shouldShowLegend=true&sort=time&spanType=all&traceQuery=&view=spans&start=1720544116827&end=1720547716827&paused=false)

**Comment by deniseli** [comment]
Also part of this: make sure metrics + logs work as intended. Add a toy one, deploy to the dev instance, make some verb calls in the dev instance, and then check datadog to make sure it all pipes all the way through. That unblocks us adding metrics/logs/traces in general.

**Comment by deniseli** [comment]
Metrics are looking pretty solid now: https://app.datadoghq.com/metric/explorer?fromUser=false&graph_layout=split&start=1722446967244&end=1722533367244&paused=false#N4Ig7glgJg5gpgFxALlAGwIYE8D2BXJVEADxQEYAaELcqyKBAC1pEbghkcLIF8qo4AMwgA7CAgg4RKUAiwAHOChASAtnADOcAE4RNIKtrgBHPJoQaUAbVBGN8qVoD6gnNtUZCKiOq279VKY6epbINiAiGOrKQdpYZAYgUJ4YThr42gDGSsgg6gi6mZaBZnHKGniqyIIIaAB0GBpYIplOmRho9aoaTgg4bTiq8miIcMAAVDwNPZn4IggAFACUIDwAulSu7niYoeFbqjsYMaXxqxsgGnIjMiDyHYgIOUk4MG27GhCZiWiicL0KZ7pX5QH5-Jz0JjKERuDxoVb8CD2TBYAGKZS-ERKdY8PiXYbiADCUmEMFuyPETigPjgIk+jmsoCk-xg2gw8kYTkU2mUNXqjWarXanTqbg4onOVF+qnEKAATAAWKjpbSEJnaAQ83ICDTfXE8IA

**Comment by deniseli** [comment]
Marking this as done now that the correctly parented traces are in prod and the new metrics+attrs are continuing to function as expected
